lib/hammer_cli_foreman/command_extensions/hosts/help/compute_resources.rb

```diff
@@ -0,0 +1,91 @@
+module HammerCLIForeman
+  module CommandExtensions
+    module Hosts
+      module Help
+        class ComputeResources < HammerCLI::CommandExtensions
+          class << self
+            attr_writer :current_providers, :add_host_specific_attrs,
+                        :attributes
+
+            def current_providers
+              @current_providers ||= HammerCLIForeman.compute_resources.keys
+              @current_providers
+            end
+
+            def add_host_specific_attrs?
+              @add_host_specific_attrs ||= true
+              @add_host_specific_attrs
+            end
+
+            def attributes
+              @attributes ||= :all
+              @attributes
+            end
+
+            def customized?
+              @customized
+            end
+
+            def customized=(boolean)
+              @customized ||= boolean
+            end
+
+            def custom(options = {})
+              new_self = Class.new(ComputeResources)
+              new_self.add_host_specific_attrs = options[:add_host_specific_attrs]
+              new_self.attributes = options[:attributes]
+              new_self.current_providers = options[:providers]
+              new_self.class_eval(help_block)
+              new_self.customized = true
+              new_self
+            end
+
+            def help_block
+              help_block = <<HELP_BLOCK
+                help do |h|
+                  h.section(_('Provider specific options')) do |h|
+                    HammerCLIForeman.compute_resources.each do |name, provider|
+                      next unless current_providers.include?(name)
+
+                      h.section(provider.name, id: name.to_sym) do |h|
+                        if add_host_specific_attrs? && defined?(provider.host_attributes)
+                          host_interfaces = provider.interface_attributes + provider.host_attributes
+                        end
+                        if %i[all volume].include?(attributes)
+                          h.section('--volume') do |h|
+                            h.list(provider.volume_attributes)
+                          end
+                        end
+                        if %i[all interface].include?(attributes)
+                          h.section('--interface') do |h|
+                            if host_interfaces
+                              h.list(host_interfaces)
```

Comment:
I don't feel that putting this code under block is the right approach, I feel like it's not readable like that, what was the reason for changing this code?

Reply:
To be able create custom help extensions, see: 
https://github.com/theforeman/hammer-cli-foreman/blob/97d96fec40e50a82de12192259fa19c6c96d7a6e/lib/hammer_cli_foreman/compute_attribute.rb#L137

https://github.com/theforeman/hammer-cli-foreman/blob/97d96fec40e50a82de12192259fa19c6c96d7a6e/lib/hammer_cli_foreman/host.rb#L304